polkadot/consensus/src/service.rs

```diff
@@ -232,12 +246,19 @@ impl Service {
 		let (signal, exit) = ::exit_future::signal();
 		let thread = thread::spawn(move || {
 			let mut core = reactor::Core::new().expect("tokio::Core could not be created");
+			let timer = ::tokio_timer::wheel()
```

Comment:
`tokio-core` implements a timer in the IO poll and supports a timer future
https://docs.rs/tokio-core/0.1.17/tokio_core/reactor/struct.Timeout.html
Can't we use that not to introduce yet another thread?

Reply:
We can file that as a TODO. It complicates error handling in the asynchronous proposer quite a lot because we want to share the result of various timeouts, but `io::Error` is not cloneable.